### PR TITLE
Generate SDKLT config automatically

### PR DESCRIPTION
**Pull request by Yi-Tseng**
The yaml file for SDKLT should be generated automatically instead of creating manually by human

### CODE REVIEW

File: stratum/hal/lib/bcm/bcm_sdklt_config_util_test.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <string>
+#include "gtest/gtest.h"
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/bcm_sdklt_config_util_test.cc]: Space between system headers and project headers

File: stratum/hal/lib/bcm/bcm_sdklt_config_util_test.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  auto sdklt_config_string = sdklt_config.ValueOrDie();
+  for (const auto& expected_string : kExpectedStrings) {
+    EXPECT_THAT(sdklt_config_string, ::testing::HasSubstr(expected_string));
+  }
+}
+
+}  // namespace bcm
+}  // namespace hal
+}  // namespace stratum
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/bcm_sdklt_config_util_test.cc]: Newline at end of file

File: stratum/hal/lib/bcm/testdata/simple_sdklt_config.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  0:
+    PC_PORT:
+      ? PORT_ID: 50
+      : PC_PHYS_PORT_ID: 49
+        ENABLE: 1
+        OPMODE: PC_PORT_OPMODE_100G
+      ? PORT_ID: 54
+      : PC_PHYS_PORT_ID: 53
+        ENABLE: 1
+        OPMODE: PC_PORT_OPMODE_100G
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/testdata/simple_sdklt_config.yml]: newline at end of file

File: stratum/hal/lib/bcm/bcm_sdklt_config_util.h [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    case BcmPort_OpMode_OPMODE_DEFAULT:
+      return kSdkltOpModeDefault;
+    case BcmPort_OpMode_OPMODE_QSGMII:
+      return kSdkltOpModeQsgmii;
+    case BcmPort_OpMode_OPMODE_GPHY:
+      return kSdkltOpModeGphy;
+    default:
+      RETURN_ERROR(ERR_INVALID_PARAM) << "Unknown operation mode " << op_mode;
+  }
+};
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/bcm_sdklt_config_util.h]: Would it make sense to move this into a protobuf? bcm.proto as enum?

File: bazel/external/yamlcpp.BUILD [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        "-Wall",
+        "-DYAML_CPP_VERSION=\\\"0.6.2\\\"",
+        "-DYAML_CPP_VERSION_MAJOR=0",
+        "-DYAML_CPP_VERSION_MINOR=6",
+        "-DYAML_CPP_VERSION_PATCH=2",
+    ],
+    deps = [
+        ":yaml_cpp_contrib"
+    ]
+)
```
Comment (pudelkoM) [inline bazel/external/yamlcpp.BUILD]: Newline at end of file

File: stratum/hal/lib/bcm/bcm_sdklt_config_util.h (hunk 4 above)
Comment (Yi-Tseng) [reply stratum/hal/lib/bcm/bcm_sdklt_config_util.h]: I believe that bcm.proto is a general structure for both SDKLT and the private SDK.
And `PC_PM_...` is for SDKLT only. (maybe I am wrong?)

Comment (pudelkoM) [reply stratum/hal/lib/bcm/bcm_sdklt_config_util.h]: You are correct. The config generator obviously depends on the SDK.

**Comment by pudelkoM** [comment]
Closed by #155

**Pull request closed without merging**